pkg/frontend/admin_hive_syncset_get.go

```diff
@@ -0,0 +1,68 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"reflect"
+
+	"github.com/go-chi/chi/v5"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/ugorji/go/codec"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) getAdminHiveSyncSet(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	syncsetname := chi.URLParam(r, "syncsetname")
+	isSyncSet := r.URL.Query().Has("isSyncSet")
+	namespace := r.URL.Query().Get("cdnamespace")
+	b, err := f._getAdminHiveSyncSet(ctx, namespace, syncsetname, isSyncSet)
+
+	if cloudErr, ok := err.(*api.CloudError); ok {
+		api.WriteCloudError(w, cloudErr)
+		return
+	}
+
+	adminReply(log, w, nil, b, err)
+}
+
+func (f *frontend) _getAdminHiveSyncSet(ctx context.Context, namespace string, syncsetname string, isSyncSet bool) ([]byte, error) {
```

Comment:
Recommend: We should not somewhere that we are using the term SyncSet for both SelectorSyncSet and plain Syncset to avoid confusion.